backend/matching-service/app/src/matching/matching.gateway.ts

```diff
@@ -0,0 +1,43 @@
+import { v4 as uuidv4 } from "uuid";
+import { Preferences, UserSocket, WaitingSocket } from "./matching.interfaces";
+import { comparePreferences } from "../utils/preferences";
+
+export class MatchingGateway {
+  public roomIdToSockets: Map<string, UserSocket[]> = new Map();
+  public waiting: WaitingSocket[] = [];
+  private roomIdToLinkWaiting: Map<string, UserSocket> = new Map();
+
+  constructor() {}
+
+  createRoom(socket: any) {
+    const roomId = uuidv4();
+    socket.join(roomId);
+    this.waiting.push({
+      roomId: uuidv4(),
+      // Hard code one existing user for testing purposes
+      preferences: {
+        difficulty: ["Medium"],
+        category: ["Array"],
+      },
+      userSocket: { user: 1, socket },
+    });
+  }
+
+  joinRandomRoom(
+    user: UserSocket,
+    preferences: Preferences,
+    socket: any
```

Comment:
Can this be typed?

Reply:
+1

Reply:
removed socket passing in 6d9abf736fe9ae8b905c31128d1f06024114dc9d